apiserver/pkg/modelservice/modelservice.go

```diff
@@ -0,0 +1,325 @@
+				Namespace: &input.Namespace,
+			}, *input.Endpoint.Auth, nil)
+			if err != nil {
+				return nil, err
+			}
+			llm.Spec.Enpoint.AuthSecret = &v1alpha1.TypedObjectReference{
+				Kind:      "Secret",
+				Name:      secret,
+				Namespace: &input.Namespace,
+			}
+		}
+
+		unstructuredLLM, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&llm)
+		if err != nil {
+			return nil, err
+		}
+
+		obj, err := c.Resource(schema.GroupVersionResource{Group: v1alpha1.GroupVersion.Group, Version: v1alpha1.GroupVersion.Version, Resource: "LLM"}).
+			Namespace(input.Namespace).Create(ctx, &unstructured.Unstructured{Object: unstructuredLLM}, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		// update auth secret owner reference
+		if input.Endpoint.Auth != nil {
+			// user obj as the owner
+			secret := common.MakeAuthSecretName(llm.Name, "LLM")
+			err := common.MakeAuthSecret(ctx, c, generated.TypedObjectReferenceInput{
+				Name:      secret,
+				Namespace: &input.Namespace,
+			}, *input.Endpoint.Auth, obj)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// create *generated.Llm
+		genLLM = obj2LLM(ctx, c, obj)
+	}
+
+	if strings.Contains(serviceType, "embedding") {
+		embedder := v1alpha1.Embedder{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      input.Name,
+				Namespace: input.Namespace,
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Embedder",
+				APIVersion: v1alpha1.GroupVersion.String(),
+			},
+			Spec: v1alpha1.EmbedderSpec{
+				CommonSpec: v1alpha1.CommonSpec{
+					DisplayName: displayName,
+					Description: description,
+				},
+				Provider: v1alpha1.Provider{
+					Enpoint: &v1alpha1.Endpoint{
+						URL: input.Endpoint.URL,
+					},
+				},
+				Type: embeddings.EmbeddingType(APIType),
+			},
+		}
+
+		unstructuredEmbedder, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&embedder)
+		if err != nil {
+			return nil, err
+		}
+		obj, err := c.Resource(schema.GroupVersionResource{Group: v1alpha1.GroupVersion.Group, Version: v1alpha1.GroupVersion.Version, Resource: "embedders"}).
+			Namespace(input.Namespace).Create(ctx, &unstructured.Unstructured{Object: unstructuredEmbedder}, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		// update auth secret owner reference
+		if input.Endpoint.Auth != nil {
+			// user obj as the owner
+			secret := common.MakeAuthSecretName(embedder.Name, "embedder")
+			err := common.MakeAuthSecret(ctx, c, generated.TypedObjectReferenceInput{
+				Name:      secret,
+				Namespace: &input.Namespace,
+			}, *input.Endpoint.Auth, obj)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		genEmbed = obj2Embedder(ctx, c, obj)
+	}
+
+	// TODO: Update auth secrets info, or is it necessary?
+	endpoint := generated.Endpoint{
+		URL:      &input.Endpoint.URL,
+		Insecure: input.Endpoint.Insecure,
+	}
+
+	// FIXME: Since ModelService isn't a CR resource, does it requires elements like 'CreationTimeStamp'?
```

Comment:
Yes .We can use the reousce creation timestamp that has been creator by this api